src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -753,9 +753,11 @@ private JObject ApplyHostConfiguration()
 
             // now the configuration has been read and applied re-create the logger
             // factory and loggers ensuring that filters and settings have been applied
+            ILoggerFactory oldLoggerFactory = _hostConfig.LoggerFactory;
             ConfigureLoggerFactory(recreate: true);
             _startupLogger = _hostConfig.LoggerFactory.CreateLogger(LogCategories.Startup);
             Logger = _hostConfig.LoggerFactory.CreateLogger(ScriptConstants.LogCategoryHostGeneral);
+            oldLoggerFactory.Dispose();
```

Comment:
Can we handle the disposal in the `ConfigureLoggerFactory` method, if `recreate` is true? Since that is where the property is updated and the logger created, it should probably be the code disposing of it as well.

Reply:
This will also change with the DI work, but with the way the host config is used throughout, it's possible we'd start seeing instances of `ObjectDisposedException` popping up as well.

Reply:
I did that at first, but that leaves a race condition where _startupLogger and Logger are both pointing to a disposed logger for a brief time. I wanted to make sure that it wasn't disposed until after we'd replaced those.

Reply:
I see. Sounds good. For V1, it doesn't warrant the work to make this change feasible and this won't be an issue in V2.